examples/elm/Main.elm

```diff
@@ -53,38 +58,51 @@ update msg model =
         ChangeName name ->
             { model | name = name }
 
+        ToggleError ->
+            { model | hasError = not model.hasError }
+
 
 onChange : (String -> Msg) -> Html.Attribute Msg
 onChange tagger =
     on "onChange" (Decode.map tagger detailTargetValueDecoder)
 
 
-
-{- IMPORTANT: We need to get the data of the original React event like
+{-| IMPORTANT: We need to get the data of the original React event like
    this : event.detail[0].target.value
 -}
-
-
 detailTargetValueDecoder : Decoder String
 detailTargetValueDecoder =
     Decode.field "detail" <|
         Decode.index 0 <|
             Decode.at [ "target", "value" ] Decode.string
 
 
+{-| Create shorthand for custom element
+-}
+customSelect : List (Attribute msg) -> List (Html msg) -> Html msg
 customSelect =
     node "custom-select"
 
 
+{-| Create shorthand for custom property. Property is easier to use
+than attribute for Boolean values.
+-}
+hasError : Bool -> Attribute Msg
+hasError =
+    property "hasError" << Encode.bool
+
+
+view : Model -> Html Msg
 view model =
     div []
         [ h3 [] [ text "Custom select component" ]
         , div []
             [ customSelect
                 [ id "customComponent"
-                , attribute "name" model.name
-                , attribute "value" model.value
-                , property "disabled" (Encode.bool model.disabled)
+                , name model.name
+                , value model.value
+                , disabled model.disabled
+                , hasError model.hasError
```

Comment:
Hmm, I thought leaving it in one file will keep the example as easy as possible. But good idea for production. Do you still think I should move it?